keras/engine/training.py

```diff
@@ -1832,7 +1832,7 @@ def generate_arrays_from_file(path):
             val_x, val_y, val_sample_weights = self._standardize_user_data(
                 val_x, val_y, val_sample_weight)
             for cbk in callbacks:
-                cbk.validation_data = val_x + [val_y, val_sample_weights]
+                cbk.validation_data = val_x + val_y + val_sample_weights
```

Comment:
Without this modification, `cbk.validation_data[1]`, which is `val_y`, is of shape [1,20,2] in `fit_generator()`, whereas it's [20,2] in `fit()`.